File: fenicia-web/src/types/basic/inventory/dashboard-data.ts
import { CategoryBreakdown } from './category-breakdown';
import { LowStockItem } from './low-stock-item';
import { SupplierBreakdown } from './supplier-breakdown';

export interface DashboardData {
    lowStockItems: LowStockItem[];
    totalCustomers: number;
    totalEmployees: number;
    totalCostValue: number;
    totalSalesValue: number;
    totalQuantity: number;
    profitPotential: number;
    categoryBreakdown: CategoryBreakdown[];
    supplierBreakdown: SupplierBreakdown[];
}

File: Src/APIs/Fenicia.Web/Components/Pages/Basic/Models/DashboardData.cs
namespace Fenicia.Web.Components.Pages.Basic.Models;

public class DashboardData
{
    public List<LowStockItem> LowStockItems { get; set; } = [];

    public int TotalCustomers { get; set; }

    public int TotalEmployees { get; set; }

    public decimal TotalCostValue { get; set; }

    public decimal TotalSalesValue { get; set; }

    public double TotalQuantity { get; set; }

    public decimal ProfitPotential { get; set; }

    public List<CategoryData> CategoryBreakdown { get; set; } = [];

    public List<SupplierData> SupplierBreakdown { get; set; } = [];
}
